services/api-server/src/simcore_service_api_server/api/routes/files.py

```diff
@@ -218,23 +213,19 @@ async def get_upload_links(
         file_size=ByteSize(client_file.filesize),
         is_directory=False,
     )
-    complete_upload_link: AnyUrl = parse_obj_as(
-        AnyUrl,
-        str(request.url_for("complete_multipart_upload", file_id=str(file_meta.id))),
-    )
-    abort_upload_link: AnyUrl = parse_obj_as(
-        AnyUrl,
-        str(request.url_for("abort_multipart_upload", file_id=str(file_meta.id))),
-    )
+    rstrip = lambda s, rs: s[: -len(rs)] if s.endswith(rs) else s
+    query = rstrip(str(upload_links.links.complete_upload.query), ":complete")
```

Comment:
is there no way to do that without a lambda? I get now what you were talking about, so the upload_links.complete ```:complete``` is parsed into the query? that is bad. is it a AnyUrl pydantic type?
```python
str.removesuffix
```
maybe?

Reply:
please note that SAN's suggestion is the right one, since `rstrip` in python behaves a bit differently.
```python
>>> "hello/me".rstrip("me")
'hello/'
>>> "hello/me".rstrip("_me")
'hello/'
>>> "hello/me".rstrip("saaame")
'hello/'
```

```python
>>> "hello/me".removesuffix("me")
'hello/'
>>> "hello/me".removesuffix("_me")
'hello/me'
```

Reply:
> please note that SAN's suggestion is the right one, since `rstrip` in python behaves a bit differently.
> 
> ```python
> >>> "hello/me".rstrip("me")
> 'hello/'
> >>> "hello/me".rstrip("_me")
> 'hello/'
> >>> "hello/me".rstrip("saaame")
> 'hello/'
> ```
> 
> ```python
> >>> "hello/me".removesuffix("me")
> 'hello/'
> >>> "hello/me".removesuffix("_me")
> 'hello/me'
> ```

This is a strange function. But I agree pylint is right.